src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java

```diff
@@ -218,6 +223,71 @@ public Double getJudgmentValue(final String judgmentsId, final String userQuery,
 
     }
 
+    @Override
+    public Map<String, Long> getRandomUbiQueries(final int n) throws IOException {
```

Comment:
My tests showed always sampling the same queries. Even when changing the size of the sample the first couple are always the same. I think running the aggregation for this sampling method as well and then doing the random sampling can make sense. We'd have an aligned distribution for each sampling method and only how we draw the sample changes.
What do you think?

Reply:
Yeah, I will try that.

Reply:
One thought about using the aggregation for random sampling is I don't think we know how many buckets there might be. At worst case the number of buckets = k which might be worse than just the manual sampling method of before. Let me tinker with it some.

Reply:
@wrigleyDan I updated the random sampler to have a seed and to collapse on the `user_query` field.  I get back different results each time now.